microcosm_sqlite/dataset.py

```diff
@@ -0,0 +1,132 @@
+"""
+Abstraction around a SQLite-based data set.
+
+"""
+from sqlalchemy.ext.declarative import declarative_base
+
+from microcosm_sqlite.context import SessionContext
+
+
+SESSION = "__session__"
+
+
+class DataSet:
+    """
+    A base class for a declarative base, representing a set of related types.
+
+    All derived types will use the same engine and session maker.
+
+    """
+    @staticmethod
```

Comment:
Most of the magic comes from the first two functions:

 - `create` generates a new declarative base class with a custom `name`, derived from `DataSet`
 - `resolve` finds the declarative base class of a derived model

Together, this lets us a) resolve configuration by the data set's name and b) save session state on the declarative base for the data set